src/main/java/io/druid/data/input/MapBasedRow.java

```diff
@@ -126,7 +126,8 @@ public long getLongMetric(String metric)
       return ((Number) metricValue).longValue();
     } else if (metricValue instanceof String) {
       try {
-        return Long.valueOf(((String) metricValue).replace(",", ""));
+        String s = ((String) metricValue).replace(",", "");
+        return s.contains(".") ? Double.valueOf(s).longValue() : Long.valueOf(s);
```

Comment:
Instead, we should simply check to make sure the string only contains +, -, digits and an optional 'L' or 'l' at the end, which are the only characters allowed according to http://docs.oracle.com/javase/7/docs/api/java/lang/Long.html#parseLong(java.lang.String)